spec/README.md

```diff
@@ -119,9 +119,10 @@ subject and unambiguously identifying the types of the [predicate]. It is a
 > content type. If this matters to you, please comment on
 > [GitHub Issue #28](https://github.com/in-toto/attestation/issues/28)
 
-`subject[*].digest` _object ([DigestSet]), required_
+`subject[*].digest` _object ([DigestSet]), optional_
 
 > Collection of cryptographic digests for the contents of this artifact.
+> A `subject` MUST have either a `digest` field or an `uri` field.
```

Comment:
I think we should more strongly state that `digest` is generally required if possible, and to only use `uri` if digest is not possible. Also that `digest` and `uri` are mutually exclusive.

(same on line 148)

Reply:
Updated the preferred form and exclusivity of digest versus uri.

Reply:
@SantiagoTorres , @mlieberman85 and I were talking about this in a slightly different context of using the attestations to be able to follow them or include locator information in them. 

For our case, i think we are thinking that `uri` and `digest` aren't mutually exclusive. For example, we want to include the digest and have the URI point to a blob store, such as `s3://..../app.spdx`, or `oci://...` or `gcs://...`. This would also potentially create a cleaner separation between rekor and its reliance on a blob store and indexing.